### ISSUE TITLE
`error: attribute 'system' missing` when using `boot.binfmt.emulatedSystems = [ "aarch64-linux" ];`

### ISSUE THREAD

**Issue by raboof** [NixOS/nixpkgs#312419]
### Describe the bug

When building NixOS with:

```
boot.binfmt.emulatedSystems = [ "aarch64-linux" ];
```

I'm now getting:

```
building the system configuration...
error:
       … while calling the 'head' builtin
         at /nix/store/0qd773b63yg8435w8hpm13zqz7iipcbs-source/lib/attrsets.nix:1575:11:
         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'
         at /nix/store/0qd773b63yg8435w8hpm13zqz7iipcbs-source/lib/modules.nix:809:9:
          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'system' missing
       at /nix/store/0qd773b63yg8435w8hpm13zqz7iipcbs-source/nixos/modules/system/boot/binfmt.nix:283:85:
          282|   config = {
          283|     boot.binfmt.registrations = builtins.listToAttrs (map (system: assert system != pkgs.system; {
             |                                                                                     ^
          284|       name = system;
       Did you mean one of mystem, systemc, systemd, syntex or systemfd?
```

on nixos-unstable at 33d1e753c82ffc557b4a585c77de43d4c922ebb5

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
already fixed in #312169

**Comment by raboof** [comment]
great!
